src/System.Security.Permissions/src/System/Security/Permissions/PrincipalPermission.cs

```diff
@@ -2,25 +2,471 @@
+
+                            idrole.m_id = "";
+                            idrole.m_role = operand.m_array[j].m_role == null ? m_array[i].m_role : operand.m_array[j].m_role;
+                            idrole.m_authenticated = operand.m_array[j].m_authenticated;
+
+                            idroles.Add(idrole);
+                        }
+                    }
+                }
+            }
+
+            if (idroles == null)
+            {
+                return null;
+            }
+            else
+            {
+                IDRole[] idrolesArray = new IDRole[idroles.Count];
+
+                IEnumerator idrolesEnumerator = idroles.GetEnumerator();
+                int index = 0;
+
+                while (idrolesEnumerator.MoveNext())
+                {
+                    idrolesArray[index++] = (IDRole)idrolesEnumerator.Current;
+                }
+
+                return new PrincipalPermission(idrolesArray);
+            }
+        }
+
+        public IPermission Union(IPermission other)
+        {
+            if (other == null)
+            {
+                return Copy();
+            }
+            else if (!VerifyType(other))
+            {
+                throw new ArgumentException(SR.Argument_WrongType, GetType().FullName);
+            }
+
+            PrincipalPermission operand = (PrincipalPermission)other;
+
+            if (IsUnrestricted() || operand.IsUnrestricted())
+            {
+                return new PrincipalPermission(PermissionState.Unrestricted);
+            }
+
+            // Now we have to do a real union
+            int combinedLength = m_array.Length + operand.m_array.Length;
+            IDRole[] idrolesArray = new IDRole[combinedLength];
+
+            int i, j;
+            for (i = 0; i < m_array.Length; ++i)
+            {
+                idrolesArray[i] = m_array[i];
+            }
+
+            for (j = 0; j < operand.m_array.Length; ++j)
+            {
+                idrolesArray[i + j] = operand.m_array[j];
+            }
+
+            return new PrincipalPermission(idrolesArray);
+
+        }
+
+        public override bool Equals(object obj)
+        {
+            IPermission perm = obj as IPermission;
+            if (obj != null && perm == null)
+                return false;
+            if (!IsSubsetOf(perm))
+                return false;
+            if (perm != null && !perm.IsSubsetOf(this))
+                return false;
+            return true;
+        }
+
+        public override int GetHashCode()
+        {
+            int hash = 0;
+            int i;
+            for (i = 0; i < m_array.Length; i++)
+                hash += m_array[i].GetHashCode();
+            return hash;
+        }
+
+        public IPermission Copy()
+        {
+            return new PrincipalPermission(m_array);
+        }
+
+        public void Demand()
+        {
+            new SecurityPermission(SecurityPermissionFlag.ControlPrincipal).Assert();
```

Comment:
We're not. This is a no-op. I'll refactor these ported files and remove this junk.